src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs

```diff
@@ -1485,20 +1485,259 @@ public async Task TypeSymbolOfTypeParameterConstraintClause3()
+        public async Task BaseList_Struct_ValidNestedType()
+        {
+            await VerifyItemExistsAsync(@"
+struct C
+{
+    class N {}
+}
+
+class B : $$
+", "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseList_NotEnum()
+        {
+            await VerifyItemIsAbsentAsync(@"
+enum C {}
+
+class B : $$
+", "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseList_NotDelegate()
+        {
+            await VerifyItemIsAbsentAsync(@"
+delegate void C();
+
+class B : $$
+", "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseList_Interface()
+        {
+            await VerifyItemExistsAsync(@"
+interface I {}
+
+class B : $$
+", "I");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListStruct_NotClass()
+        {
+            await VerifyItemIsAbsentAsync(@"
+class C {}
+
+struct B : $$
+", "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListStruct_Class_ValidNestedType()
+        {
+            await VerifyItemExistsAsync(@"
+class C
+{
+    interface I {}
+}
+
+struct B : $$
+", "C");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListStruct_Interface()
+        {
+            await VerifyItemExistsAsync(@"
+interface I {}
+
+struct B : $$
+", "I");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListInterface_Interface()
+        {
+            await VerifyItemExistsAsync(@"
+interface I2 {}
+
+interface I1 : $$
+", "I2");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListInterface_NotClass()
+        {
+            await VerifyItemIsAbsentAsync(@"
+class N {}
+
+interface I : $$
+", "N");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListInterface_Class_ValidNestedType()
+        {
+            await VerifyItemExistsAsync(@"
+class N
+{
+    interface I2 {}
+}
+
+interface I : $$
+", "N");
+        }
+
+        [WorkItem(60935, "https://github.com/dotnet/roslyn/issues/60935")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListInterface_NotSelf()
+        {
+            await VerifyItemIsAbsentAsync(@"
+interface I : $$
+", "I");
+        }
+
+        [WorkItem(60935, "https://github.com/dotnet/roslyn/issues/60935")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task BaseListInterface_NotSelfEvenWithNestedTypes()
+        {
+            await VerifyItemIsAbsentAsync(@"
+interface I : $$
+{
+    interface I2 {}
+}", "I");
+        }
```

Comment:
Please add several tests that include things like writing the second base type. Or if the base type is partially written. Or if you're supplying type arguments to the base type